File: libraryBe/src/main/java/com/example/libraryBe/entity/Fine.java
package com.example.libraryBe.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "fines")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Fine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "book_loan_id")
    private BookLoan bookLoan;

    private Double amount; // Số tiền phạt
    private String description; // Lý do phạt

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
}

File: library-fe-2025/src/types/fine.ts
export interface Fine {
    id: number;
    amount: number;
    description: string;
    createdAt: string;
    updatedAt: string;
    user: {
        id: number;
        username: string;
        name: string;
    };
    bookLoan?: {
        id: number;
        bookCopy: {
            book: {
                title: string;
                coverUrl: string;
            }
        }
    };
}